llvm/tools/sycl-post-link/SYCLDeviceRequirements.cpp

```diff
@@ -42,11 +49,15 @@ void llvm::getSYCLDeviceRequirements(
     for (const Function &F : MD.getModule()) {
       if (const MDNode *MDN = F.getMetadata(MDName)) {
         for (size_t I = 0, E = MDN->getNumOperands(); I < E; ++I) {
-          // Don't put internal aspects (with negative integer value) into the
-          // requirements, they are used only for device image splitting.
-          auto Val = ExtractIntegerFromMDNodeOperand(MDN, I);
-          if (Val >= 0)
-            Values.insert(Val);
+          if (std::string(MDName) == "sycl_used_aspects") {
+            // Don't put internal aspects (with negative integer value) into the
+            // requirements, they are used only for device image splitting.
+            auto Val = ExtractSignedIntegerFromMDNodeOperand(MDN, I);
```

Comment:
Where do these internal aspects with negative values come from? I am unfamiliar with them

Reply:
I added them in scope of this PR: https://github.com/intel/llvm/pull/10140